llama_index/indices/query/base.py

```diff
@@ -2,20 +2,33 @@
 
 import logging
 from abc import ABC, abstractmethod
-from typing import List, Optional, Sequence
+from typing import Any, Dict, List, Optional, Sequence
 
 from llama_index.callbacks.base import CallbackManager
 from llama_index.indices.query.schema import QueryBundle, QueryType
+from llama_index.prompts.base import BasePromptTemplate
+from llama_index.prompts.mixin import PromptMixin
 from llama_index.response.schema import RESPONSE_TYPE
 from llama_index.schema import NodeWithScore
 
 logger = logging.getLogger(__name__)
 
 
-class BaseQueryEngine(ABC):
+class BaseQueryEngine(PromptMixin):
     def __init__(self, callback_manager: Optional[CallbackManager]) -> None:
         self.callback_manager = callback_manager or CallbackManager([])
 
+    def _get_prompts(self) -> Dict[str, Any]:
+        """Get prompts."""
+        return {}
+
+    def _get_prompt_modules(self) -> Dict[str, Any]:
+        """Get prompt modules."""
+        return {}
+
+    def _update_prompts(self, **prompts: BasePromptTemplate) -> None:
+        """Update prompts."""
```

Comment:
nit: need to `pass` here?

Reply:
i think make format might have taken it out, can redo